File: exercise/kotlin/week02-exercise2/src/main/kotlin/BookStore.kt
class BookStore {
    private var inv: ArrayList<Book> = ArrayList()

    fun addBook(title: String?, author: String?, copies: Int) {
        if (title != null && author != null && copies > 0) {
            var foundBook: Book? = null
            for (book in inv) {
                if (book.title == title && book.author == author) {
                    foundBook = book
                    break
                }
            }
            if (foundBook != null) {
                foundBook.addCopies(copies)
            } else {
                inv.add(Book(title, author, copies))
            }
        }
    }

    fun sellBook(title: String, author: String, copies: Int) {
        for (book in inv) {
            if (book.title == title && book.author == author) {
                book.removeCopies(copies)
                if (book.copies <= 0) {
                    inv.remove(book)
                }
                break
            }
        }
    }
}

File: exercise/c#/week02-object-calisthenics/BookStore/BookStore.cs
namespace BookStore;

public class BookStore
{
    private readonly List<Book> _inv = new();

    public void AddBook(string? title, string? author, int copies)
    {
        if (title != null && author != null && copies > 0)
        {
            Book? foundBook = null;
            foreach (var book in _inv)
            {
                if (book.Title == title && book.Author == author)
                {
                    foundBook = book;
                    break;
                }
            }

            if (foundBook != null)
            {
                foundBook.AddCopies(copies);
            }
            else
            {
                _inv.Add(new Book(title, author, copies));
            }
        }
    }

    public void SellBook(string title, string author, int copies)
    {
        foreach (var book in _inv)
        {
            if (book.Title == title && book.Author == author)
            {
                book.RemoveCopies(copies);
                if (book.Copies <= 0)
                {
                    _inv.Remove(book);
                }

                break;
            }
        }
    }
}